src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php

```diff
@@ -108,7 +108,7 @@ public function getProxyCode(Definition $definition)
      */
     private function getProxyClassName(Definition $definition)
     {
-        return preg_replace('/^.*\\\\/', '', $definition->getClass()).'_'.substr(hash('sha256', spl_object_hash($definition).$this->salt), -7);
+        return preg_replace('/^.*\\\\/', '', $definition->getClass()).'_'.substr(hash('sha256', $definition->getClass().$this->salt), -7);
```

Comment:
Does this mean we only have a maximum of one proxy per class? 🤔

Reply:
does it mean anything to have more that one proxy per class? do you have an example?

Reply:
you will get only one proxy *class* per class. You can still have multiple instance.

If you look at the code, the *only* part of the Definition impact the generated class is the class name. So multiple definitions can reuse the same proxy class safely.

Reply:
You are right, this makes sense :)